server.c

```diff
@@ -2749,15 +2800,28 @@ server_child(struct nsd *nsd)
 	 * connections.
 	 */
 	if (nsd->server_kind & NSD_SERVER_TCP) {
+		int child = nsd->this_child->child_num;
 		tcp_accept_handler_count = numifs;
 		tcp_accept_handlers = region_alloc_array(server_region,
 			numifs, sizeof(*tcp_accept_handlers));
 
-		for (i = from; i < numifs; i++) {
-			struct tcp_accept_handler_data *data =
-				&tcp_accept_handlers[i-from];
-			memset(data, 0, sizeof(*data));
```

Comment:
You should not remove this memset, perhaps move it inside the `add_tcp_handler` function, but you cannot delete it, it would make a fixed bug reappear.

Reply:
Restored `memset`